superset-frontend/src/cccs-viz/plugins/plugin-chart-dataset-explorer/src/plugin/transformProps.ts

```diff
@@ -0,0 +1,363 @@
+   */
+  const {
+    datasource,
+    hooks,
+    width,
+    height,
+    rawFormData: formData,
+    queriesData,
+  } = chartProps;
+  const {
+    boldText,
+    headerFontSize,
+    headerText,
+    emitFilter,
+    principalColumns,
+    query_mode,
+    column_state,
+  }: CccsGridQueryFormData = { ...DEFAULT_FORM_DATA, ...formData };
+  const data = queriesData[0].data as TimeseriesDataRecord[];
+  const agGridLicenseKey = queriesData[0].agGridLicenseKey as String;
+
+  const default_group_by: any[] = [];
+  const enable_row_numbers: boolean = true;
+  const jump_action_configs: string[] = []
+  const enable_json_expand: boolean = false 
+
+  const include_search: boolean = true;
+  const page_length: number = 50;
+  const enable_grouping: boolean = true;
+
+
+  const { setDataMask = () => {}, setControlValue } = hooks;
+  const columns = datasource?.columns as Column[];
+  const metrics = datasource?.metrics as Metric[];
+
+  const columnTypeMap = new Map<string, string>();
+  columns.reduce(function (columnMap, column: Column) {
+    // @ts-ignore
+    const name = column.column_name;
+    // @ts-ignore
+    columnMap[name] = column.type;
+    return columnMap;
+  }, columnTypeMap);
+
+  // Map of column advanced types, key is column name, value is column type
+  const columnAdvancedTypeMap = new Map<string, string>();
+  columns.reduce(function (columnMap, column: Column) {
+    // @ts-ignore
+    const name = column.column_name;
+    // @ts-ignore
+    columnMap[name] = (
+      (column.advanced_data_type as string) ?? ''
+    ).toUpperCase();
+    return columnMap;
+  }, columnAdvancedTypeMap);
+
+  // Map of verbose names, key is column name, value is verbose name
+  const columnVerboseNameMap = new Map<string, string>();
+  columns.reduce(function (columnMap, column: Column) {
+    // @ts-ignore
+    const name = column.column_name;
+    // @ts-ignore
+    columnMap[name] = column.verbose_name;
+    return columnMap;
+  }, columnVerboseNameMap);
+
+  // Map of column descriptions, key is column name, value is the description
+  const columnDescriptionMap = new Map<string, string>();
+  columns.reduce(function (columnMap, column: Column) {
+    // @ts-ignore
+    const name = column.column_name;
+    // @ts-ignore
+    columnMap[name] = column.description;
+    return columnMap;
+  }, columnDescriptionMap);
+
+  // Map of verbose names, key is metric name, value is verbose name
+  const metricVerboseNameMap = new Map<string, string>();
+  metrics.reduce(function (metricMap, metric: Metric) {
+    // @ts-ignore
+    const name = metric.metric_name;
+    // @ts-ignore
+    metricMap[name] = metric.verbose_name;
+    return metricMap;
+  }, metricVerboseNameMap);
+
+  // Key is column advanced type, value is renderer name
+  const rendererMap = {
+    IPV4: 'ipv4ValueRenderer',
+    IPV6: 'ipv6ValueRenderer',
+    DOMAIN: 'domainValueRenderer',
+    COUNTRY: 'countryValueRenderer',
+    JSON: 'jsonValueRenderer',
+  };
```

Comment:
Maybe this isn't a conversation to be had here, however, there are 28 advanced data types in DataHub (which will eventually be added to Superset). Some of them will not need to have any special rendering, but some will. Is there anywhere we can put them so that this viz and the hogwarts table can both make use of them?